wicket-extensions/src/main/java/org/apache/wicket/extensions/ajax/markup/html/modal/ModalDialog.js

```diff
@@ -0,0 +1,424 @@
+	}
+
+	/** Focuses the first element inside the modal */
+	function focusDefaultFocusable(container) {
+		var matches = findFocusable(container);
+		var first = matches.not(".modal-dialog-close").first();
+		if (first.length > 0) {
+			first.focus();
+		} else {
+			matches.first().focus();
+		}
+	}
+
+	/**
+	 * Finds and clicks the close button inside the modal. Returns true if
+	 * button was found.
+	 */
+	function findAndClickCloseButton(container) {
+		var matches = container.find(CLOSE_SELECTOR).filter(":visible");
+		if (matches.length > 0) {
+			matches.first().click();
+			return true;
+		} else {
+			return false;
+		}
+	}
+
+	//
+	// BEHAVIORS
+	//
+	// Behaviors are event listeners that get called from open and close
+	// methods, they allow various aspects of code such as focus management and
+	// aria attribute management to be decoupled from each other making the
+	// overall code cleaner and easier to maintain.
+	//
+	// The structure of a behavior is an object with the following properties:
+	// initialize: function()
+	// - called before first modal is opened
+	// destroy: function()
+	// - called after last modal is closed
+	// prepare: function(overlayElement, contentElement, data)
+	// - called after overlay dom is constructed, but before it is inserted
+	// - into main dom
+	// open: function(overlayElement, contentElement, data)
+	// - called after overlayElement is inserted into main dom
+	// close: function(overlayElement, contentElement, data)
+	// - called after overlayElement is removed from main dom
+
+	//Scroll settings to remember for ios scroll to top issue. Currently, ios allows body 
+	//scrolling unless body is set to position: fixed, which causes the window to scroll to top.
+    var scrollTop;
+
+	/** Behavior that appends a css class to body as long as any modal is open */
+	var appendBodyClassBehavior = {
+
+		initialize : function() {
+			var body = $("body");
+			body.addClass("modal-dialog-open modal-dialog-no-scroll");
+
+			scrollTop = $(window).scrollTop();
+
+            if (!!navigator.platform && /iPad|iPhone|iPod/.test(navigator.platform)) {
+                body.addClass("modal-dialog-open-ios");
+            }
+		},
+		terminate : function() {
+			$("body").removeClass("modal-dialog-open modal-dialog-no-scroll modal-dialog-open-ios");
+            if (!!navigator.platform && /iPad|iPhone|iPod/.test(navigator.platform)) {
+            	$(window).scrollTop(scrollTop);
+            }
+		}
+	};
+
+	/**
+	 * Behavior that memorizes the focussed element when dialog is opened, and
+	 * returns focus to it when dialog is closed
+	 */
+	var returnFocusOnCloseBehavior = {
+		open : function(overlay, element, data) {
+			data.opener = document.activeElement;
+			if (data.options.validate) {
+				if (!data.opener || $(data.opener).is("body")) {
+					data.options.console.error("Error saving focused element when opening the modal, it is either none or body: ",
```

Comment:
What is the idea behind `options.console` ? To be able to use either `window.console`, `Wicket.Log` or a third party one ?

Reply:
yes, and also to be able to suppress any warnings/errors in production build